source/common/upstream/upstream_impl.h

```diff
@@ -764,25 +765,45 @@ class ClusterInfoImpl : public ClusterInfo,
   clusterType() const override {
     return cluster_type_;
   }
-  const absl::optional<envoy::config::cluster::v3::Cluster::RoundRobinLbConfig>&
+  OptRef<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig>
   lbRoundRobinConfig() const override {
-    return lb_round_robin_config_;
+    if (lb_round_robin_config_ == nullptr) {
+      return absl::nullopt;
+    } else {
+      return *lb_round_robin_config_;
+    }
```

Comment:
nit: reduce code complexity by reducing nesting (here and below)
This can be rewritten by:
```
if (...) {
  return ;
}
return ;
```

Reply:
Done